processing/src/main/java/org/apache/druid/query/filter/vector/ReadableVectorMatch.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.filter.vector;
+
+import javax.annotation.Nullable;
+
+/**
+ * The result of calling {@link VectorValueMatcher#match}.
+ *
+ * @see VectorMatch, the implementation, which also adds some extra mutation methods.
+ */
+public interface ReadableVectorMatch
+{
+  /**
+   * Returns an array of indexes into the current batch. Only the first "getSelectionSize" are valid.
+   *
+   * Even though this array is technically mutable, it is very poor form to mutate it if you are not the owner of the
+   * VectorMatch object.
+   */
+  int[] getSelection();
```

Comment:
I got it, with _getSelection_ we expecting to trigger **HotSpot vectorization** elsewhere, that way we could leave the method but I'd indicate the purpose in the method comment.

Reply:
Yep, that is it. I added a comment.

Reply:
Could you mention `HotSpot vectorization` term in the comment?

Reply:
I added this sentence:

> Potential optimizations could include making it easier for the JVM to use CPU-level vectorization, avoid method calls, etc.

I'm not 100% sure that the CPU level vectorization will _actually_ be triggered, but it could theoretically be, which is the point. Even if it isn't today, it might be in future JVMs. (And it might be today, I just haven't checked.)